File: example/java/data-authorization/src/main/java/org/babyfish/jimmer/sql/example/database/DatabaseInitializer.java
package org.babyfish.jimmer.sql.example.database;

import org.babyfish.jimmer.sql.JSqlClient;
import org.babyfish.jimmer.sql.example.Context;
import org.babyfish.jimmer.sql.runtime.Initializer;

import java.io.IOException;
import java.io.InputStreamReader;
import java.io.Reader;
import java.sql.SQLException;
import java.sql.Statement;

public class DatabaseInitializer implements Initializer, Context {

    @Override
    public void initialize(JSqlClient sqlClient) throws Exception {
        TRANSACTION_MANAGER.execute(() -> {
            try {
                String text = loadSql();
                initDatabase(text);
            } catch (IOException | SQLException ex) {
                throw new RuntimeException("Cannot initialize h2 database", ex);
            }
            return null;
        });
    }

    private String loadSql() throws IOException {
        StringBuilder builder = new StringBuilder();
        char[] buf = new char[1024];
        try (Reader reader = new InputStreamReader(DatabaseInitializer.class.getClassLoader().getResourceAsStream("database.sql"))) {
            int len;
            while ((len = reader.read(buf)) != -1) {
                builder.append(buf, 0, len);
            }
        }
        return builder.toString();
    }

    private void initDatabase(String text) throws SQLException {
        String[] arr = text.split("\\s*;\\s*");
        for (String sql : arr) {
            Statement statement = TRANSACTION_MANAGER.currentConnection().createStatement();
            statement.executeUpdate(sql);
        }
    }
}

File: example/kotlin/jimmer-cloud-kt/store-service/src/main/kotlin/org/babyfish/jimmer/example/cloud/kt/store/cfg/DatabaseInitializer.kt
package org.babyfish.jimmer.example.cloud.kt.store.cfg

import org.babyfish.jimmer.sql.kt.KSqlClient
import org.babyfish.jimmer.sql.kt.cfg.KInitializer
import org.springframework.stereotype.Component
import java.io.InputStreamReader
import javax.sql.DataSource

@Component
class DatabaseInitializer(private val dataSource: DataSource) : KInitializer {

    override fun initialize(dsl: KSqlClient) {
        dataSource.connection.use { con ->
            val inputStream = DatabaseInitializer::class.java
                .classLoader
                .getResourceAsStream("store.sql") ?: throw RuntimeException("no `store.sql`")
            InputStreamReader(inputStream).use { reader ->
                con.createStatement().execute(reader.readText())
            }
        }
    }
}